src/engine/dataapi/DataStore.java

```diff
@@ -29,18 +29,24 @@ public DataStore() {
 	
 	/**
 	 * Actually receives an individual data storage request
+	 * @throws Exception 
 	 */
 	@Override
-	public void receiveDataStoreRequest(ProtoDataStream newData) {
+	public void receiveDataStoreRequest(ProtoDataStream newData) throws Exception {
```

Comment:
Need to validate data and print stack trace, not just throw an arbitrary exception.